services/lambda-label-successful-prs/CIBot.py

```diff
@@ -0,0 +1,199 @@
+
+    def _get_github_object(self):
+        """
+        This method returns github object initialized with Github personal access token
+        """
+        github_obj = Github(self.github_personal_access_token)
+        return github_obj
+
+    def _find_all_open_prs(self, github_obj):
+        """
+        This method identifies all the open PRs in the given repository
+        It returns a Paginated list of Pull Request objects sorted in
+        descending order of created date i.e. latest PR first
+        :param github_obj
+        """
+        repo = github_obj.get_repo(self.repo)
+        pulls = repo.get_pulls(state='open', sort='created', direction='desc')
+        return pulls
+
+    def _get_jenkins_obj(self):
+        """
+        This method returns an object of Jenkins instantiated using username, password
+        """
+        return Jenkins(self.jenkins_url, username=self.jenkins_username, password=self.jenkins_password)
+
+    def _get_checks(self):
+        """
+        This method finds out all the checks that are currently supported as part of CI
+        """
+        # for now hardcoding list of checks
+        # ideally use Jenkins API to query list of checks and parse it (bit complicated)
+        all_checks = ['clang', 'edge', 'centos-cpu', 'centos-gpu', 'windows-cpu', 'windows-gpu', 'miscellaneous', 'unix-cpu', 'unix-gpu', 'website', 'sanity']
+        return all_checks
+
+    def _is_successful(self, pr):
+        """
+        This method checks if all the pipelines corresponding to a latest build of the PR are successful
+        :param pr PullRequest object
+        """
+        # get jenkins object
+        jenkins_obj = self._get_jenkins_obj()
+
+        # get PR number
+        pr_number = pr.number
+
+        # get a list of checks
+        # Github says "status checks"
+        # Jenkins refers to it as a job name
+        checks = self._get_checks()
+
+        # iterate over the checks
+        for check in checks:
+            # Jenkins Job name
+            name = "mxnet-validation/" + check + "/PR-" + str(pr_number)
+
+            try:
+                # get Jenkins job specific object
+                job = jenkins_obj[name]
+            except Exception:
+                logging.error(f'Unknown Job Exception')
+                self.prs_unknown_job += 1
+                return False
+            # get status of the latest build of the job
+            try:
+                latest_build = job.get_last_build()
+                status = latest_build.get_status()
+                logging.info(f'Latest build status for PR {pr_number} is {status}')
+                if(status.upper() == 'FAILURE'):
+                    self.prs_status_fail += 1
+                    return False
+            except Exception:
+                logging.error(f'No Build Exception')
+                self.prs_no_build += 1
+                return False
+        logging.info(f'All 11 builds succeeded! Happy case!')
```

Comment:
11?

Reply:
Ya i hardcoded it. But with the github API we don't have to do this.